include/geos/operation/polygonize/BuildArea.h

```diff
@@ -0,0 +1,76 @@
+/**********************************************************************
```

Comment:
I think @dr-jts made some improvements in the JTS polygonizer (excluding holes from output) that may make this redundant. I see now problem including it now and replacing the implementation later, though.

Reply:
No, Polygonizer has not changed recently.  I might have done some experimental work on a BuildArea equivalent though.  Will have to look for the code. So probably not relevant to this PR.

Reply:
@dr-jts I was thinking of this: https://github.com/dbaston/jts-svn-history/commit/31e500b2eb0b4ebdea8b1cdedb1f20a3559ba983#diff-9c4ebe19717a7d64b755b588909103a7

Reply:
@dbaston ah right, had forgotten about that.  And I see a bit of cleanup needed there too...

As you say, can always port this JTS code later and replace internals of BuildArea.